smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/metric/fetcher/InotifyEventFetcher.java

```diff
@@ -203,13 +228,95 @@ public void run() {
     private final InotifyEventApplier applier;
     private final QueueFile queueFile;
     private long lastId;
+    private SmartConf conf;
+    private List<String> ignoreList;
 
     public EventApplyTask(NamespaceFetcher namespaceFetcher, InotifyEventApplier applier,
         QueueFile queueFile, long lastId) {
       this.namespaceFetcher = namespaceFetcher;
       this.queueFile = queueFile;
       this.applier = applier;
       this.lastId = lastId;
+      this.conf = new SmartConf();
+      this.ignoreList = getIgnoreDirFromConfig();
+    }
+
+    public EventApplyTask(NamespaceFetcher namespaceFetcher, InotifyEventApplier applier,
+        QueueFile queueFile, long lastId, SmartConf conf) {
+      this.namespaceFetcher = namespaceFetcher;
+      this.queueFile = queueFile;
+      this.applier = applier;
+      this.lastId = lastId;
+      this.conf = conf;
+      this.ignoreList = getIgnoreDirFromConfig();
+    }
+
+    public List<String> getIgnoreDirFromConfig() {
+      String ignoreDirs = this.conf.get(SmartConfKeys.SMART_IGNORE_DIRS_KEY);
+      List<String> ignoreList;
+      if (ignoreDirs == null || ignoreDirs.equals("")) {
+        ignoreList = new ArrayList<>();
+      } else {
+        ignoreList = Arrays.asList(ignoreDirs.split(","));
```

Comment:
Add tail '/' to ignoreList here instead of doing it every time in 'fetchPathInIgnoreList'